eng/SourceBuildPrebuiltBaseline.xml

```diff
@@ -14,5 +14,21 @@
     <UsagePattern IdentityGlob="Microsoft.CodeAnalysis.CSharp.CodeStyle/*" />
     <UsagePattern IdentityGlob="Microsoft.CodeAnalysis.VisualBasic.CodeStyle/*" />
     <UsagePattern IdentityGlob="Microsoft.Net.Compilers.Toolset/*" />
+
+    <!-- Roslyn's source-build CI builds both NetPrevious and NetCurrent. This 7.0 ref pack shows up as
+         prebuilt only for the repo CI build but not full source-build. -->
+    <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/7.0*" />
+
+    <!-- The repo has a dependency on this package that shows up as a prebuilt in the repo's CI but in full
+         source-build this will be overridden. -->
+    <UsagePattern IdentityGlob="Microsoft.DiaSymReader/2.0.0" />
+
+    <!-- Roslyn's source-build CI builds both NetPrevious and NetCurrent. This 7.0 ref pack shows up as
+         prebuilt only for the repo CI build but not full source-build. -->
+    <UsagePattern IdentityGlob="Microsoft.NETCore.App.Ref/7.0*" />
+
+    <!-- The repo has a dependency on this package that shows up as a prebuilt in the repo's CI but in full
+         source-build this will be overridden. -->
+    <UsagePattern IdentityGlob="System.CommandLine/2.0.0*" />
```

Comment:
Per conversation with @mmitche - the System.CommandLine and Microsoft.DiaSymReader dependencies in Version.Details.xml should be adorned with the sourcebuild attribute.  The value in this is two fold.  First, it will remove the need to baseline these because the intermediate will get loaded (if this version exists) and eliminate the prebuilt.  Second, as a result of the previous, it will build with the SB variant of the dependency which may be slightly different than the MSFT version.

Reply:
Looks like the System.CommandLine 2.0.0-beta4.23307.1 intermediate doesn't exist so we will still need this exclusion.

Reply:
It should: https://dev.azure.com/dnceng/internal/_build/results?buildId=2196074&view=artifacts&pathAsName=false&type=publishedArtifacts

Reply:
Missing NuGet feed?

Reply:
![image](https://github.com/dotnet/roslyn/assets/8725170/157ed7c5-a4e9-4001-be32-afb3bb465fe0)

dotnet-libraries-transport, and note different version number for the SB intermediate

Reply:
So how is the intermediate version supposed to be referenced in the source?